UserScrape/crawler.py

```diff
@@ -122,108 +137,213 @@ def login(self) -> CrawlResult:
+        # 3. The ui is in the wrong language
+        if len(delete_buttons)>0:
+            delete_buttons[0].click()
+        self.__log_info('deleted first video from history')
+
+
+    def delete_history(self):
+        self.driver.get('https://www.youtube.com/feed/history')
+        self.__log_info('before deleting full history')
+
+        
+        messages = self.driver.find_elements_by_xpath("//*[@id='message']")
+        # if there are not videos in the history a text appears that says 'no videos here' but apparently there is a second, hidden, message with the same
+        # id on the page. So instead of checking whether this element exists we differentiate between 1 message (there are videos in the history) and 
+        # two messages (there are no videos in the history)
+        if len(messages)==1:
+            # Clear all watch history
+            # Gesamten Wiedergabeverlauf löschen
+            delete_buttons = self.driver.find_element_by_xpath("//*[@aria-label = 'Gesamten Wiedergabeverlauf löschen']").click()
+
+            # todo replace with english terms
+            # CLEAR WATCH HISTORY
+            # WIEDERGABEVERLAUF LÖSCHEN
+            confirm_button = self.driver.find_element_by_xpath("//*[@aria-label = 'WIEDERGABEVERLAUF LÖSCHEN']").click()
+        self.__log_info('deleted history')
+
+
+    def _get_seconds(self, duration: str):
+        # helper function to correctly parse the time from the info bar
+        if len(duration) > 5:
+            duration_time = datetime.strptime(duration, "%H:%M:%S")
+        else:
+            duration_time = datetime.strptime(duration, "%M:%S")
+        return (duration_time-datetime(1900, 1, 1)).total_seconds()
+
+    def watch_video(self, videoId: str):
+        """[summary]
+        starts video, skips any ads in the beginning and then watches video for an amount of time that is dependent on video length
+        Currently we are ignoring ads that appear in the middle of the video. At the moment we are assuming that watching an ad also contributes
+        to the watchtime of the video itself
+        Arguments:
+            videoId {str} -- The id of the video to be watched
+        """
+        seshPath = self.path_session()
+
+        
+        self.driver.get("https://www.youtube.com/watch?v=" + videoId)
+        # wait until video is loaded
+        playbutton = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.XPATH,
+                                            '//*[@class="ytp-play-button ytp-button"]'))
+        )
+        # todo: in the end this will be only english
+        if playbutton.get_attribute('title') in ["Play (k)", "Wiedergabe (k)"]:
+            playbutton.click()
+        # unfortunately the ad loads slower than the player so we wait here to be sure we detect the ad if any appears
+        time.sleep(1)
+        # we store any advertisements that appear
+        advertisements = {videoId: []}
```

Comment:
I am not sure what the best format is for storing the advertisers as well as the watch time. How should we best store this?